api/admin/controller/individual_admin_settings.py

```diff
@@ -57,25 +55,51 @@ def process_post(self):
         else:
             roles = []
 
-        roles_error = self.handle_roles(user, admin, roles, settingUp)
+        roles_error = self.handle_roles(admin, roles, settingUp)
         if roles_error:
             return roles_error
 
         password = flask.request.form.get("password")
-        self.handle_password(password, admin, is_new, user)
+        self.handle_password(password, admin, is_new)
 
         return self.response(admin, is_new)
 
     def check_permissions(self, admin, settingUp):
         """Before going any further, check that the user actually has permission
          to create/edit this type of admin"""
-        # User must be a sitewide library manager in order to create/edit
-        # a sitewide library manager.
-        if admin.is_sitewide_library_manager() and not settingUp:
-            self.require_sitewide_library_manager()
-        # User must be a system admin in order to create/edit a system admin.
-        if admin.is_system_admin() and not settingUp:
-            self.require_system_admin()
+        if not settingUp:
+            user = flask.request.admin
+            if not user.id == admin.id:
```

Comment:
I'm not sure if you actually need this check. (You might, I'd have to think through it)